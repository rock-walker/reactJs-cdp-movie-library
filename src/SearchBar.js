import React, {Component} from 'react';
import './Components.css';

class SearchBar extends React.Component {
	render() {
		return (
			<input type="text" className="searchBar"/>
		);
	}
}

export default SearchBar;