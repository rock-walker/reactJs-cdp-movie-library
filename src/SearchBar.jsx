import React, {Component} from 'react';
import './Components.css';

class SearchBar extends Component {
	render() {
		return (
            <div>
                <h1 className="App-title">Find your movie</h1>
                <input type="text" className="searchBar"/>
            </div>
        );
    }
}

export default SearchBar;