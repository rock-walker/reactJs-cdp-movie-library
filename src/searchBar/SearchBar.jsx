import React, {Component} from 'react';
import '../searchBar/searchBar.css';

class SearchBar extends Component {
    constructor(props){
        super(props);
        this.state = {searchText: ''};
        this.handleSearchText = this.handleSearchText.bind(this);
    }

    handleSearchText (event) {
        this.setState({
            searchText: event.target.value
        });
    }

	render() {
		return (
            <div>
                <h1 className="App-title">Find your movie</h1>
                <input type="text" 
                    value={this.state.searchText} 
                    onChange={this.handleSearchText} 
                    className="searchBar"/>
            </div>
        );
    }
}

export default SearchBar;