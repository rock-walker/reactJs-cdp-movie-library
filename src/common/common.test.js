import React from 'react';
import {mount, configure} from 'enzyme';
import ErrorBoundary from '../common/ErrorBoundary';
import SearchBar from '../searchBar/SearchBar';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });


function DefectiveChild() {
    throw new Error('failed to display');
    return <div>Failed</div>;
}

describe('Error boundary', () => {
    test('catch error', () => {
        //const spy = expect.spyOn(ErrorBoundary.prototype, 'componentDidCatch');
        const component = renderer.create(<ErrorBoundary><DefectiveChild/></ErrorBoundary>);
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot();
        //expect(ErrorBoundary.prototype.componentDidCatch).toHaveBeenCalled();
    })

    test('display component correctly', () => {
        const component = renderer.create(<ErrorBoundary><SearchBar/></ErrorBoundary>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})