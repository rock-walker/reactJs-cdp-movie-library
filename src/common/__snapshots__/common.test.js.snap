// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error boundary catch error 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`Error boundary display component correctly 1`] = `
<div>
  <h1
    className="App-title"
  >
    Find your movie
  </h1>
  <input
    className="searchBar"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
